hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenMintHandler.java

```diff
@@ -214,6 +211,7 @@ private List<Long> mintNonFungible(
 
         // Change the supply on token
         changeSupply(token, treasuryRel, metadataCount, FAIL_INVALID, accountStore, tokenStore, tokenRelStore);
+        treasuryAccount = accountStore.get(treasuryRel.accountIdOrThrow());
```

Comment:
Why assign `treasuryAccount` twice?

Reply:
We are updating `treasuryAccount` in `changeSupply` and were not getting latest account from store again.
This was causing `numPositiveBalances` to be set incorrectly.

Reply:
We should add a comment. I was confused, too.

Reply:
Yes, will add now.

Reply:
Added comments. Thanks @mhess-swl @netopyr